buildSrc/src/main/java/org/opensearch/gradle/zipplugin/ZipPublish.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.gradle.zipplugin;
+
+import java.util.*;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.publish.PublishingExtension;
+import org.gradle.api.publish.maven.MavenPublication;
+import org.gradle.api.publish.maven.plugins.MavenPublishPlugin;
+import java.nio.file.Path;
+import org.gradle.api.Task;
+
+public class ZipPublish implements Plugin<Project> {
+    private Project project;
+
+    public final static String EXTENSION_NAME = "zipmavensettings";
+    public final static String PUBLICATION_NAME = "mavenzip";
+    public final static String STAGING_REPO = "zipstaging";
+    public final static String MAVEN_ZIP_PUBLISH_POM_TASK = "generatePomFileForMavenzipPublication";
+    public final static String LOCALMAVEN = "publishToMavenLocal";
+    public final static String LOCAL_STAGING_REPO_PATH = "/build/local-staging-repo";
+    public String zipDistributionLocation = "/build/distributions/";
+
+    private void configMaven(Project project) {
+        final Path buildDirectory = project.getRootDir().toPath();
+        project.getPluginManager().apply(MavenPublishPlugin.class);
+        project.getExtensions().configure(PublishingExtension.class, publishing -> {
+            publishing.repositories(repositories -> {
+                repositories.maven(maven -> {
```

Comment:
Why do we need `LOCAL_STAGING_REPO_PATH`? I assume we should use appropriate Maven repository (local or not)

Reply:
@reta `LOCAL_STAGING_REPO_PATH = "/build/local-staging-repo";`, this is a file system path, that we consider as a local staging repo, for maven (in terms on maven terminology) its just a publication repo, so considering this `local-staging-repo` which is added across build scripts I have added `LOCAL_STAGING_REPO_PATH`, which denotes that its a filesystem path for locally staging the artifacts.

Reply:
I see, thanks @prudhvigodithi , not much we can do about it ...